VideoLocker/src/main/java/org/edx/mobile/task/Task.java

```diff
@@ -22,32 +27,50 @@
 
     private ProgressBar progressBar;
 
-    private WeakReference<TaskProcessCallback> taskProcessCallback;
+    @Nullable
+    private WeakReference<TaskProgressCallback> progressCallback;
+    @Nullable
+    private WeakReference<TaskMessageCallback> messageCallback;
 
     protected final Handler handler = new Handler();
     protected final Logger logger = new Logger(getClass().getName());
 
     @Inject
     protected IEdxEnvironment environment;
-    
+
     public Task(Context context) {
         super(context);
 
-        if ( context instanceof TaskProcessCallback ){
+        if (context instanceof TaskProcessCallback) {
             setTaskProcessCallback((TaskProcessCallback) context);
         }
     }
-    
+
     public void setProgressDialog(ProgressBar progressBar) {
         this.progressBar = progressBar;
     }
 
-    public void setTaskProcessCallback(TaskProcessCallback callback){
-        this.taskProcessCallback = new WeakReference<TaskProcessCallback>(callback);
+    public void setTaskProcessCallback(@Nullable TaskProcessCallback callback) {
+        setProgressCallback(callback);
+        setMessageCallback(callback);
+    }
+
+    public void setProgressCallback(@Nullable TaskProgressCallback callback) {
+        progressCallback = callback == null ? null : new WeakReference<>(callback);
+    }
+
+    public void setMessageCallback(@Nullable TaskMessageCallback callback) {
+        messageCallback = callback == null ? null : new WeakReference<>(callback);
+    }
+
+    @Nullable
+    private TaskProgressCallback getProgressCallback() {
+        return this.progressCallback == null ? null : this.progressCallback.get();
```

Comment:
I guess we can remove the unnecessary `this.` accessor from these getters as well.